### PR TITLE
improve emoji detection logic and add test cases

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Move the emoji String extension from Wire-iOS (https://github.com/wireapp/wire-ios/pull/2787).

String.containsEmoji method is merged with revised logic and more test case for unicode 11.0 and ancient language are added.

### CODE REVIEW

File: Source/Validation/ZMPhoneNumberValidator.m [hunk 1]
```diff
@@ -50,7 +50,7 @@ @implementation ZMPhoneNumberValidator
 
 + (BOOL)validateValue:(inout id *)ioPhoneNumber error:(out NSError **)outError
 {
-    if ([(*ioPhoneNumber) length] < 1) {
+    if ([(NSString *)(*ioPhoneNumber) length] < 1) {
```
Comment (billypchan) [inline Source/Validation/ZMPhoneNumberValidator.m]: fix compiler error of ioPhoneNumber.length ambiguity issue

**Comment by billypchan** [comment]
@zenkins release minor
